core/src/html/dimensions.rs

```diff
@@ -0,0 +1,360 @@
+//! CSS dimension types
+use gc_arena::Collect;
+use std::cmp::{max, min, Ord};
+use std::ops::{Add, AddAssign, Sub};
+use swf::{Rectangle, Twips};
+
+/// A type which represents the top-left position of a layout box.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Collect)]
+#[collect(require_static)]
+pub struct Position<T> {
+    x: T,
+    y: T,
+}
+
+impl<T> Default for Position<T>
+where
+    T: Default,
+{
+    fn default() -> Self {
+        Self {
+            x: T::default(),
+            y: T::default(),
+        }
+    }
+}
+
+impl<T> From<(T, T)> for Position<T> {
+    fn from(pair: (T, T)) -> Self {
+        Self {
+            x: pair.0,
+            y: pair.1,
+        }
+    }
+}
+
+impl<T> Position<T> {
+    pub fn set_x(&mut self, x: T) {
+        self.x = x;
+    }
+
+    pub fn set_y(&mut self, y: T) {
+        self.y = y;
+    }
+}
+
+impl<T> Position<T>
+where
+    T: Clone,
+{
+    pub fn x(&self) -> T {
+        self.x.clone()
+    }
+
+    pub fn y(&self) -> T {
+        self.y.clone()
+    }
+}
+
+impl<T> Add for Position<T>
+where
+    T: Add<T, Output = T>,
+{
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        Self {
+            x: self.x + rhs.x,
+            y: self.y + rhs.y,
+        }
+    }
+}
+
+impl<T> AddAssign for Position<T>
+where
+    T: AddAssign,
+{
+    fn add_assign(&mut self, rhs: Self) {
+        self.x += rhs.x;
+        self.y += rhs.y;
+    }
+}
+
+/// A type which represents the size of a layout box.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Collect)]
+#[collect(require_static)]
+pub struct Size<T> {
+    width: T,
+    height: T,
+}
+
+impl<T> Default for Size<T>
+where
+    T: Default,
+{
+    fn default() -> Self {
+        Self {
+            width: T::default(),
+            height: T::default(),
+        }
+    }
+}
+
+impl<T> From<(T, T)> for Size<T> {
+    fn from(pair: (T, T)) -> Self {
+        Self {
+            width: pair.0,
+            height: pair.1,
+        }
+    }
+}
+
+#[allow(dead_code)]
+impl<T> Size<T>
+where
+    T: Clone,
+{
+    pub fn width(&self) -> T {
+        self.width.clone()
+    }
+
+    pub fn height(&self) -> T {
+        self.height.clone()
+    }
+}
+
+impl<T> From<Position<T>> for Size<T> {
+    fn from(size: Position<T>) -> Self {
+        Self {
+            width: size.x,
+            height: size.y,
+        }
+    }
+}
+
+/// A type which represents the offset and size of text-orientation-relative
+/// boxes.
+///
+/// For the purposes of internationalization, the two axes are called "primary"
+/// and "secondary", and the directions are "leading" for the direction of text
+/// flow and "trailing" for the opposite direction. The meaning of these axis
+/// and directions are determined by the `WritingMode` of a given `LayoutBox`.
+///
+/// To obtain concretely-oriented dimensions, use `into_screen_space` to
+/// transform the coordinates into a pair of top, left, right, and bottom
+/// coordinates.
```

Comment:
*Extremely* out of date, back before I realized none of non-TLF had any sort of internationalization support whatsoever.

On a related note, I'm not entirely sure if we need this type anymore or if I should just roll it into `Rectangle` since it wound up becoming very similar to that type...

Reply:
Thanks! I do think some of these dimension could be rolled together/used elsewhere in the modules, but that can be a later task